### PR TITLE
cleanup(rules): update container_entrypoint (1/n)

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind feature

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

> /area maturity-stable

> /area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:

Update `container_entrypoint` to reflect changes in behavior based on recent work by @Andreagit97 to improve the process tree logic in the underlying `libs`.

There will be follow up PRs to update the behavior across runtimes. This PR is focused on `containerd` runtime.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -2007,7 +2007,7 @@
 # when we lose events and lose track of state.
 
 - macro: container_entrypoint
-  condition: (not proc.pname exists or proc.pname in (runc:[0:PARENT], runc:[1:CHILD], runc, docker-runc, exe, docker-runc-cur))
+  condition: (not proc.pname exists or proc.pname in (runc:[0:PARENT], runc:[1:CHILD], runc, docker-runc, exe, docker-runc-cur, containerd-shim-runc-v2, systemd))
```
Comment (incertum) [inline rules/falco_rules.yaml:2010]: Include `systemd` for robustness or fail safe as in odd edge cases this could happen with the current logic.

Comment (darryk10) [reply rules/falco_rules.yaml]: what about `podman`?

Comment (incertum) [reply rules/falco_rules.yaml]: Absolutely, we need to check for each supported container engine. Was hoping it's gonna be a team effort. For this first PR it's only definitely valid for containerd, rest we do in as many follow up PRs as we need.

**Review by darryk10 — APPROVED** [review]

**Review by Andreagit97 — APPROVED** [review]
/approve
Makes sense, thanks!

**Pull request merged**